tests/kafkatest/services/trogdor/degraded_network_fault_spec.py

```diff
@@ -23,7 +23,7 @@ class DegradedNetworkFaultSpec(TaskSpec):
     Degrades the network so that traffic on a subset of nodes has higher latency
     """
 
-    def __init__(self, start_ms, duration_ms, node_specs):
+    def __init__(self, start_ms, duration_ms, node_specs={}):
```

Comment:
In python, "{}" defines a mutable dictionary.  Since this is mutable, subsequent code can mutate it and effectively change the default parameter for everyone else. 
 See https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument

Rather than doing this, you should make the default "None" (which is not mutable) and subsequently check for None and initialize to a new empty dictionary.

Reply:
That is indeed astonishing 😖. I'll just remove this param from the constructor. It currently has no usage anyways.